Add experts map section to compiler team page (#286)
rust-lang/rustc-guide: src/compiler-team.md

- The `t-compiler` stream on [the Zulip instance](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler)
- The `compiler` channel on the [rust-lang discord](https://discord.gg/rust-lang)

## Expert map

We maintain a [list of areas of the compiler and who on the team knows a

> bit about that particular area](https://github.com/rust-lang/compiler-team/blob/master/experts/MAP.md).

Reviewer: This link is a 404 for me.

Reviewer 2: It is intended to be until a compiler-team PR lands.

Reviewer: For posterity: https://github.com/rust-lang/compiler-team/pull/40

Reviewer 3: I would modify this paragraph in the same way that I suggested on rust-lang/compiler-team#40, I think -- i..e., to give a better idea of how the list is meant to be used, and invite people to ask questions.

> This may be useful when you're working on some specific thing and may

Reviewer 4: ```suggestion
This is useful when you're working on some specific thing and
```

> need somebody to help you.

Reviewer 2: ```suggestion
need somebody to help you or when looking for reviewers for a PR.
```
